File: ruby/can_place_flowers.rb
# frozen_string_literal: true

# @param {Integer[]} flowerbed
# @param {Integer} n
# @return {Boolean}
def can_place_flowers(flowerbed, n)
  count = 0

  (0...flowerbed.size).each do |i|
    if flowerbed[i] == 0 && (i == 0 || (flowerbed[i - 1]).zero?) && (i == flowerbed.size - 1 || (flowerbed[i + 1]).zero?)
      count += 1
      flowerbed[i] = 1
    end

    return true if count >= n
  end

  false
end

File: python/can_place_flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        count = 0
        size = len(flowerbed)
        i = 0

        while i < size:
            if flowerbed[i] == 0 and (i == 0 or flowerbed[i - 1] == 0) and (i == size - 1 or flowerbed[i + 1] == 0):
                count += 1
                flowerbed[i] = 1
                i += 1

            if count >= n:
                return True

            i += 1

        return False
